### PR TITLE
updated rllib examples to conform to rllib verison 0.5.0 DO NOT MERGE

### PR DESCRIPTION
**Pull request by eugenevinitsky**
- All examples are updated to work in 0.5.0
- Merging blocked until ARS PR is merged and eugenevinitsky/Ray is updated

### CODE REVIEW

**Comment by eugenevinitsky** [comment]
Note, this is expected to fail the tests for now.

File: examples/rllib/cooperative_merge.py [hunk 1]
```diff
@@ -130,7 +130,7 @@
 )
 
 if __name__ == "__main__":
-    ray.init(num_cpus=PARALLEL_ROLLOUTS, redirect_output=False)
+    ray.init(num_cpus=PARALLEL_ROLLOUTS+1, redirect_output=False)
```
Comment (AboudyKreidieh) [inline examples/rllib/cooperative_merge.py]: should we start using `N_CPUS` instead of `PARALLEL_ROLLOUTS`? So that people don't ask for 8 for example when their laptop don't have 9 cpus

File: examples/rllib/figure_eight.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
 # number of rollouts per training iteration
 N_ROLLOUTS = 20
 # number of parallel workers
-PARALLEL_ROLLOUTS = 2
+PARALLEL_ROLLOUTS = 20
```
Comment (AboudyKreidieh) [inline examples/rllib/figure_eight.py]: keep at 2 if someone wants to run this locally

File: flow/benchmarks/figureeight0.py [hunk 3]
```diff
@@ -17,7 +17,7 @@
 from flow.scenarios.figure8.figure8_scenario import ADDITIONAL_NET_PARAMS
 
 # time horizon of a single rollout
-HORIZON = 1500
+HORIZON = 300
```
Comment (AboudyKreidieh) [inline flow/benchmarks/figureeight0.py]: undo this

File: flow/benchmarks/figureeight0.py [hunk 4]
```diff
@@ -51,7 +51,7 @@
 
     # sumo-related parameters (see flow.core.params.SumoParams)
     sumo=SumoParams(
-        sim_step=0.1,
+        sim_step=0.5,
```
Comment (AboudyKreidieh) [inline flow/benchmarks/figureeight0.py]: undo this

File: flow/benchmarks/rllib/es_runner.py [hunk 5]
```diff
@@ -1,9 +1,56 @@
...
+            "env": env_name,
+            "config": {
+                **config
+            },
+            "checkpoint_freq": 5,
+            "max_failures": 999,
+            "stop": {"training_iteration": 500},
+            # "upload_dir": "s3://bucket"
+        },
+    })
```
Comment (AboudyKreidieh) [inline flow/benchmarks/rllib/es_runner.py]: thanks for adding this

File: flow/benchmarks/rllib/ars_runner.py [hunk 6]
```diff
@@ -10,37 +10,36 @@
...
 # use this to specify the environment to run
-from flow.benchmarks.figureeight2 import flow_params
+from flow.benchmarks.figureeight0 import flow_params
 
 # number of rollouts per training iteration
-N_ROLLOUTS = 25
+N_ROLLOUTS = 2
 # number of parallel workers
-PARALLEL_ROLLOUTS = 25
+PARALLEL_ROLLOUTS = 2
```
Comment (AboudyKreidieh) [inline flow/benchmarks/rllib/ars_runner.py]: are these changes intentional? maybe undo

File: scripts/ray_autoscale.yaml [hunk 7]
```diff
@@ -69,17 +69,14 @@ worker_nodes:
 file_mounts: {
 # path to your repo and the desired branch name
 #"/tmp/path": "<PATH TO LEARNING TRAFFIC>/.git/refs/heads/<BRANCH NAME>",
-#"/tmp/ray_autoscaler_key": "<PATH TO RAY_AUTOSCALER_KEY>",
-"/tmp/path": "/Users/eugenevinitsky/Desktop/Research/Bayen/Code/flow/.git/refs/heads/raytesting",
-"/tmp/ray_autoscaler_key": "/Users/eugenevinitsky/.ssh/ray-autoscaler_1_us-west-1.pem",
+"/tmp/path": "/Users/eugenevinitsky/Desktop/Research/Bayen/Code/flow/.git/refs/heads/update_rllib_version",
```
Comment (AboudyKreidieh) [inline scripts/ray_autoscale.yaml]: do not keep personal implementations here

File: examples/rllib/cooperative_merge.py (hunk 1 above)
Comment (eugenevinitsky) [reply examples/rllib/cooperative_merge.py]: Yeah, that's clearer. I'll change them all to say that.

**Comment by eugenevinitsky** [comment]
Comments addressed.
